File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.cpp
#include <array>

class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int l = 0;
        int r = 0;
        int res = 0;
        unordered_set<char> characters{};

        while (r < s.length())
        {
            while (l < r && characters.contains(s[r]))
            {
                characters.erase(s[l]);
                ++l;
            }
            characters.insert(s[r]);
            res = max(res, r - l + 1);
            ++r;
        }  
        return res;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        HashSet<int> set = new HashSet<int>(nums);
        int res = 0;
        foreach (int n in nums) {
            if (set.Contains(n - 1)) {
                continue;
            }
            set.Add(n);
            int next = n;
            while (set.Contains(next)) {
                ++next;
            }
            res = Math.Max(res, next - n);
        }
        return res;
    }
}
